cognite/client/data_classes/data_modeling/query.py

```diff
@@ -107,28 +124,14 @@ def _load(cls, resource: dict[str, Any], cognite_client: AsyncCogniteClient | No
             limit=resource.get("limit"),
         )
 
-    def _validate_for_sync(self, name: str) -> None:
-        if self.sort:
-            raise ValueError(f"Select expression '{name}' has sort set, which is not allowed for the sync endpoint.")
-        if self.limit is not None:
-            raise ValueError(f"Select expression '{name}' has limit set, which is not allowed for the sync endpoint.")
-
-
-class Query(CogniteObject):
-    r"""Query allows you to do advanced queries on the data model.
-
-    Args:
-        with_ (dict[str, ResultSetExpression]): A dictionary of result set expressions to use in the query. The keys are used to reference the result set expressions in the select and parameters.
-        select (dict[str, Select]): A dictionary of select expressions to use in the query. The keys must match the keys in the with\_ dictionary. The select expressions define which properties to include in the result set.
-        parameters (dict[str, PropertyValue] | None): Values in filters can be parameterised. Parameters are provided as part of the query object, and referenced in the filter itself.
-        cursors (Mapping[str, str | None] | None): A dictionary of cursors to use in the query. These are for pagination purposes, for example, in the sync endpoint.
-    """
 
+class QueryBase(CogniteObject, ABC):
+    "Abstract base class for normal queries and sync queries"
```

Comment:
nit, PEP 257 wants triple quotes also for one-liners:
```suggestion
    """Abstract base class for normal queries and sync queries"""
```